src/AEG/AEGTree.ts

```diff
@@ -115,21 +114,23 @@ export class AEGTree {
 
         if (incomingNode instanceof AtomNode) {
             if (otherNode instanceof AtomNode) {
-                return (incomingNode as AtomNode).rect.overlaps((otherNode as AtomNode).rect);
+                return (incomingNode as AtomNode).Rectangle.overlaps(
+                    (otherNode as AtomNode).Rectangle
+                );
             } else {
                 //the case where otherNode is the sheet is handled in canInsert()
                 //and all child.ellipse[i] will never be null. this is the reason for ! below
 
-                ellipse1 = (otherNode as CutNode).ellipse!;
-                return (incomingNode as AtomNode).rect.overlaps(ellipse1);
+                ellipse1 = (otherNode as CutNode).Ellipse!;
+                return (incomingNode as AtomNode).Rectangle.overlaps(ellipse1);
             }
         } else {
             if (otherNode instanceof AtomNode) {
-                ellipse1 = (incomingNode as CutNode).ellipse!;
-                return ellipse1.overlaps((otherNode as AtomNode).rect);
+                ellipse1 = (incomingNode as CutNode).Ellipse!;
+                return ellipse1.overlaps((otherNode as AtomNode).Rectangle);
             } else {
-                ellipse1 = (incomingNode as CutNode).ellipse!;
-                ellipse2 = (otherNode as CutNode).ellipse!;
+                ellipse1 = (incomingNode as CutNode).Ellipse!;
+                ellipse2 = (otherNode as CutNode).Ellipse!;
                 return ellipse1.overlaps(ellipse2);
             }
```

Comment:
This logic can be simplified into 1-2 lines by directly calling the AEGUtils overlaps function, good use case for it.